bundledata_test.go

```diff
@@ -166,6 +166,32 @@ relations:
 			{"mysql:foo", "mediawiki:bar"},
 		},
 	},
+}, {
+	about: "legacy bundle with services instead of applications",
+	data: `
+services:
+    wordpress:
+        charm: wordpress
+    mysql:
+        charm: mysql
+        num_units: 1
+relations:
+    - ["wordpress:db", "mysql:db"]
+`,
+	expectedBD: &charm.BundleData{
+		Applications: map[string]*charm.ApplicationSpec{
+			"wordpress": {
+				Charm: "wordpress",
+			},
+			"mysql": {
+				Charm:    "mysql",
+				NumUnits: 1,
+			},
+		},
+		Relations: [][]string{
+			{"wordpress:db", "mysql:db"},
+		},
+	},
 }}
```

Comment:
Let's have a test for a bundle that specifies both service and applications, please.
(I think that should be an error).